src/tiledb/cloud/geospatial/ingestion.py

```diff
@@ -0,0 +1,1182 @@
+    resampling: Optional[str] = "bilinear",
+    res: Tuple[float, float] = None,
+    verbose: bool = False,
+    trace: bool = False,
+    log_uri: Optional[str] = None,
+    unit_testing: bool = False,
+) -> None:
+    """
+    Ingests geospatial point clouds, geometries and images into TileDB arrays
+
+    :param dataset_uri: dataset URI
+    :param dataset_type: Classifier, one of pointcloud, raster or geometry
+    :param acn: Access Credentials Name (ACN) registered in TileDB Cloud (ARN type),
+        defaults to None
+    :param config: config dictionary, defaults to None
+    :param namespace: TileDB-Cloud namespace, defaults to None
+    :param register_name: name to register the dataset with on TileDB Cloud,
+        defaults to None
+    :param search_uri: URI to search for geospatial dataset files, defaults to None
+    :param pattern: Unix shell style pattern to match when searching for files,
+        defaults to None
+    :param ignore: Unix shell style pattern to ignore when searching for files,
+        defaults to None
+    :param dataset_list_uri: URI with a list of dataset URIs, defaults to None
+    :param max_files: maximum number of URIs to read/find,
+        defaults to None (no limit)
+    :param compression_filter: serialized tiledb filter,
+        defaults to None
+    :param workers: number of workers for dataset ingestion, defaults to MAX_WORKERS
+    :param batch_size: batch size for dataset ingestion, defaults to BATCH_SIZE
+    :param tile_size: for rasters this is the tile (block) size
+        for the merged destination array, defaults to 1024
+    :param chunk_size: for point cloud this is the PDAL chunk size, defaults to 1000000
+    :param nodata: NODATA value for raster merging
+    :param resampling: string, resampling method,
+        one of None, bilinear, cubic, nearest and average
+    :param res: Tuple[float, float], output resolution in x/y
+    :param verbose: verbose logging, defaults to False
+    :param trace: bool, enabling log tracing, defaults to False
+    :param log_uri: log array URI
+    :param unit_testing: bool internal flag for unit testing the code flow
+    """
+
+    logger = get_logger_wrapper(verbose)
+
+    kwargs = {}
+    fn = None
+
+    if dataset_list_uri:
+        sources = read_uris(
+            dataset_list_uri,
+            dataset_type=dataset_type,
+            log_uri=log_uri,
+            config=config,
+            max_files=max_files,
+        )
+
+    def pointcloud_match(f):
+        return os.path.splitext(f)[-1].lstrip(".").lower() in ["las", "laz"]
+
+    def raster_match(f):
+        try:
+            rasterio.driver_from_extension(f)
+            return True
+        except ValueError:
+            return False
+
+    def geometry_match(f):
+        try:
+            fiona.drvsupport.driver_from_extension(f)
+            return True
+        except ValueError:
+            return False
+
+    if dataset_type == "pointcloud":
```

Comment:
You could simplify code by defining a dict that will store for each key (`datatype`) the functions used for `metadata` and `udfs` and then the rest of the code will be less repeatable and you will need to define just the different kwargs.

Reply:
👍